backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java

```diff
@@ -107,4 +114,25 @@ public ProposalRevisionEmail requestChanges(
     private boolean hasFeedback(ProposalRevisionDto proposalRevisionDto) {
         return proposalRevisionDto.getFeedback() != null;
     }
+
+    /**
+     * This method find the proposal in the database and checks if the status is REVIEW_PENDING or ENROLLMENT_CLOSED for
+     * publish. Otherwise, a BadProposalStatusException with response status 400 will be throw.
+     * @param idProposal : The id of the proposal to be checked and updated.
+     * @return result with the proposal person email and proposal title.
+     */
+    public ChangeStatusToPublishedResult changeStatusToPublished(String idProposal) {
+        JpaProposal proposal = jpaProposalRepository.findByNaturalId(idProposal).orElseThrow(EntityNotFoundException::new);
+        String status = proposal.getStatus().getName().toUpperCase();
+        if (REVIEW_PENDING.toString().equals(status) || ENROLLMENT_CLOSED.toString().equals(status)) {
+            JpaProposalStatus jpaProposalStatus = JpaProposalStatus.builder()
+                    .id(ProposalStatus.PUBLISHED.getId())
+                    .name("PUBLISHED").build();
+            jpaProposalRepository.changeStatusToPublished(idProposal, jpaProposalStatus);
+
+            return new ChangeStatusToPublishedResult(jpaCredentialRepository.getContactPersonEmail(idProposal), proposal.getTitle());
```

Comment:
Use the ESAL Id retrieve above to make a request to to jpaContactPersonRepository.findByEsaldWithCredentials(esalId)

You can now get the email from the credentials in JpaContactPerson :-)

Reply:
Got it! Thank you!